lib/b64.c

```diff
@@ -185,11 +186,15 @@ jose_b64_dec(const json_t *i, void *o, size_t ol)
 
     if (json_unpack((json_t *) i, "s%", &b64, &len) < 0)
         return SIZE_MAX;
-
-    if (!o)
-        return b64_dlen(len);
-
-    return jose_b64_dec_buf(b64, len, o, ol);
+    size_t padding = 0;
+    for (size_t idx = 0; idx < len; idx++) {
```

Comment:
Isn't padding only at the end of the string ?
In that case I would rather:
size_t idx = len - 1;
while (b64[idx] == '=') {
    idx--;
}
len = idx + 1;